specutils/spectra/xrayspectrum1d.py

```diff
@@ -0,0 +1,542 @@
+import logging
+
+import numpy as np
+from astropy import units as u
+from astropy.io import fits
+from .spectrum1d import Spectrum1D
+
+__all__ = ['XraySpectrum1D', 'ARF', 'RMF']
+
+# For dealing with varied unit string choices
+EV   = ['eV', 'ev']
+KEV  = ['kev', 'keV']
+ANGS = ['angs', 'Angs', 'Angstrom', 'angstrom', 'Angstroms', 'angstroms', 'A', 'a']
+
+def _unit_parser(unit_string):
+    """
+    A function for matching unit strings to the correct Astropy unit
+    """
+    if unit_string in EV:
+        return u.eV
+    if unit_string in KEV:
+        return u.keV
+    if unit_string in ANGS:
+        return u.angstrom
+
+class XraySpectrum1D(Spectrum1D):
+    """
+    Spectrum properties specific to X-ray data
+
+    Parameters
+    ----------
+    bin_lo : numpy.ndarray
+        The left edges for bin values
+
+    bin_hi : numpy.ndarray
+        The right edges for bin values
+
+    bin_unit : string OR astropy.units.Unit
+        Unit for the bin values
+
+    counts : numpy.ndarray
+        Counts histogram for the X-ray spectrum
+
+    exposure : float
+        Exposure time for the dataset
+
+    arf : specutils.ARF or string, default None
+        Strings will be passed to ARF.__init__
+        All other input types are stored as arf attribute
+
+    rmf : specutils.RMF or string, default None
+        Strings will be passed to RMF.__init__
+        All other input types are stored as rmf attribute
+
+    rest_value : astropy.units.Quantity, default 0 Angstrom
+        See Spectrum1D rest_value input
+
+    Attributes
+    ----------
+    Inherits from Spectrum1D object.
+    The following inputs are stored as additional attributes.
+
+    bin_lo
+
+    bin_hi
+
+    counts
+
+    exposure
+
+    arf
+
+    rmf
+    """
+    def __init__(self, bin_lo, bin_hi, bin_unit, counts, exposure,
+                 arf=None, rmf=None, rest_value=0.0 * u.angstrom, **kwargs):
+        try:
+            axis_unit = u.Unit(bin_unit)
+        except:
```

Comment:
Could you catch the _specific_ error you expect here (`ValueError`, I believe, in the case where the unit cannot be parsed)?